sharding/utils/marshal_test.go

```diff
@@ -84,5 +83,127 @@ func TestSerializeAndDeserializeblob(t *testing.T) {
 			t.Errorf("Error Serializing blobs at index %d, the serialized and deserialized versions are not the same:\n\n %v \n\n %v \n\n %v", i, blob, serializedblob, raw)
 		}
 	}
+}
+
+func TestDeserializeSkipEvm(t *testing.T) {
```

Comment:
The existing tests can cover a lot of cases but I think they could cause some problems down the road. We should avoid testing against objects created with a random number generator because they can fail randomly and in unexpected ways. They also don't test specific cases, meaning that when they do fail, its incredibly unclear what caused the failure. Finally they take a relatively long time to run, and we want to be able to run the entire unit test suite as quickly as possible!

Reply:
Agreed, the reason I used random number generator was to make it easier to uncover edge cases. Currently there is not standardised tests yet for serialisation but in the future there will probably be standardised json tests so we would not be needing to use RNG.